intersect_test.go

```diff
@@ -270,6 +270,26 @@ func TestWithout(t *testing.T) {
 	is.IsType(nonempty, allStrings, "type preserved")
 }
 
+func TestWithoutBy(t *testing.T) {
+	t.Parallel()
```

Comment:
Can also add this in the test , This gives a more real life example:


```
func main() {
    // Example usage
    users := []User{
        {ID: 1, Name: "Alice"},
        {ID: 2, Name: "Bob"},
        {ID: 3, Name: "Charlie"},
    }

    // Exclude users with IDs 2 and 3
    excludedIDs := []int{2, 3}

    // Extract function to get the user ID
    extractID := func(user User) int {
        return user.ID
    }

    // Filtering users
    filteredUsers := WithoutBy(users, extractID, excludedIDs...)

    // Output the filtered users
    for _, user := range filteredUsers {
        fmt.Println(user.Name)
    }
}
```

Reply:
I added a intersect_example_test.go file to add this example. Please review it again.